src/index.js

```diff
@@ -135,13 +135,22 @@ function matchesSelector (element, ast) {
       if (node.value === ' ') {
         // walk all ancestors
         const precedingNodes = getLastNonCombinatorNodes(nodes.slice(0, i))
-        const ancestor = getFirstMatchingAncestor(element, precedingNodes)
+        const prePrecedingNodes = nodes.slice(0, i - precedingNodes.length)
```

Comment:
After thinking about it a bit more, I think the right name is not `prePrecedingNodes` but something more like `allNodesUpToLastCombinatorNodes` or something.

It's hard to describe, but for example you'd have:

- `nodes`: `[div, .container, >, div, :nth-child, ' ', select]`
- `precedingNodes`: `[div, :nth-child]`
- `allNodesUpToLastCombinatorNodes`: `[div, .container, >]`

... where the `element` is `<select>` and the `ancestor` is a`<div>`.

I'd propose renaming these to:

- `precedingNodes`: `lastNonCombinatorNodes`
- `prePrecedingNodes`: `nodesBeforeLastNonCombinatorNodes`